### PR TITLE
Update CI.yaml for Py3.12 and drop Py3.8

### LINKED ISSUE #97: Check Python 3.12 compatibility with develop version of MDAnalysis

### PR DESCRIPTION
**Pull request by hmacdope**
Fixes #97

## Status
- [x] Ready to go

### CODE REVIEW

**Comment by hmacdope** [comment]
@orionarcher the blocker for Python 12 here is the pandas upper pin. What was the reasoning behind that to begin with before I change it?

**Comment by hmacdope** [comment]
Added missing setuptools.

File: pyproject.toml [hunk 1]
```diff
@@ -29,10 +29,11 @@ keywords = [
 
 dependencies = [
     'numpy>=1.20.0',
-    'pandas<2.1',
+    'pandas',
     'mdanalysis>=2.0.0',
     'pytest',
     'matplotlib',
+    'setuptools',
```
Comment (IAlibay) [inline pyproject.toml:36]: Huh setuptools as a runtime dependency is really rare - what are you using it for? Usually it goes under build time dependencies.

P.S. if you drop versioneer you could drop setup.py completely I think

Comment (hmacdope) [reply pyproject.toml:36]: using `pkg_resources` in python 3.12 has an explicit dependency on setuptools now. IIRC there is a replacement but I was too lazy to search it out.

Comment (hmacdope) [reply pyproject.toml:36]: I always mess up the `versioningit` setup for some reason.

Comment (IAlibay) [reply pyproject.toml:36]: Ah yeah - I'd drop `pkg_resources`, best thing is to use importlib. Here's the giant MDA example 😅 : https://github.com/MDAnalysis/mdanalysis/blob/develop/testsuite/MDAnalysisTests/datafiles.py

Comment (IAlibay) [reply pyproject.toml:36]: re: versioningit setup - here's the recent change we did for waterdynamics, hopefully should give you all the things that need changing: https://github.com/MDAnalysis/waterdynamics/pull/28/files

File: devtools/conda-envs/test_env.yaml [hunk 2]
```diff
@@ -7,8 +7,9 @@ dependencies:
   - pip
   - MDAnalysis>=2.0.0
   - scipy
+  - setuptools
   - numpy
-  - pandas<2.1
+  - pandas
```
[the remark this answers is not in the document]
Comment (orionarcher) [reply devtools/conda-envs/test_env.yaml]: Good catch

**Comment by orionarcher** [comment]
Thanks for the maintenance work here @hmacdope. The pin to 2.2 should kill two birds with one stone. Will merge when tests are passing.

**Comment by orionarcher** [comment]
Agreed and will do!

**Pull request merged**